crates/types/src/signer.rs

```diff
@@ -96,6 +97,31 @@ impl SignMode {
         }
     }
 
+    /// Returns the verifying key of the block producer, if any
+    pub fn verifying_key(&self) -> anyhow::Result<Option<k256::ecdsa::VerifyingKey>> {
+        match self {
+            SignMode::Unavailable => Ok(None),
+            SignMode::Key(secret_key) => {
+                let secret: k256::SecretKey = secret_key.expose_secret().as_ref().into();
+                let public_key = secret.public_key();
+
+                Ok(Some(public_key.into()))
+            }
+
+            #[cfg(feature = "aws-kms")]
+            SignMode::Kms {
+                cached_public_key_bytes,
+                ..
+            } => {
+                use k256::pkcs8::DecodePublicKey;
+
+                let k256_public_key =
+                    k256::PublicKey::from_public_key_der(cached_public_key_bytes)?;
+                Ok(Some(k256_public_key.into()))
```

Comment:
q: why don't we just set the public key into the enum variant instead of its raw form?  we won't have to recalculate it when `verifying_key()` is called every time.

as i understand the code, `make_payload()` is leading to this function being called repeatedly

Reply:
It is more question to @Dentosal =) Performance is not critical in `make_payload` because we call this function twice each 12 seconds